magic-nix-cache/src/flakehub.rs

```diff
@@ -137,24 +139,53 @@ pub async fn init_cache(
 
     // Periodically refresh JWT in GitHub Actions environment
     if environment.is_github_actions() {
-        if let super::FlakeHubAuthSource::Netrc(path) = auth_method {
-            // NOTE(cole-h): This is a workaround -- at the time of writing, GitHub Actions JWTs are only
-            // valid for 5 minutes after being issued. FlakeHub uses these JWTs for authentication, which
-            // means that after those 5 minutes have passed and the token is expired, FlakeHub (and by
-            // extension FlakeHub Cache) will no longer allow requests using this token. However, GitHub
-            // gives us a way to repeatedly request new tokens, so we utilize that and refresh the token
-            // every 2 minutes (less than half of the lifetime of the token).
-            let netrc_path_clone = path.to_path_buf();
-            let initial_github_jwt_clone = flakehub_password.clone();
-            let flakehub_cache_server_clone = flakehub_cache_server.to_string();
-            let api_clone = api.clone();
-
-            tokio::task::spawn(refresh_github_actions_jwt_worker(
-                netrc_path_clone,
-                initial_github_jwt_clone,
-                flakehub_cache_server_clone,
-                api_clone,
-            ));
+        match auth_method {
+            super::FlakeHubAuthSource::Netrc(path) => {
+                // NOTE(cole-h): This is a workaround -- at the time of writing, GitHub Actions JWTs are only
+                // valid for 5 minutes after being issued. FlakeHub uses these JWTs for authentication, which
+                // means that after those 5 minutes have passed and the token is expired, FlakeHub (and by
+                // extension FlakeHub Cache) will no longer allow requests using this token. However, GitHub
+                // gives us a way to repeatedly request new tokens, so we utilize that and refresh the token
+                // every 2 minutes (less than half of the lifetime of the token).
+                let netrc_path_clone = path.to_path_buf();
+                let initial_github_jwt_clone = flakehub_password.clone();
+                let flakehub_cache_server_clone = flakehub_cache_server.to_string();
+                let api_clone = api.clone();
+
+                tokio::task::spawn(refresh_github_actions_jwt_worker(
+                    netrc_path_clone,
+                    initial_github_jwt_clone,
+                    flakehub_cache_server_clone,
+                    api_clone,
+                ));
+            }
+            crate::FlakeHubAuthSource::DeterminateNixd => {
+                // NOTE(cole-h): This is a workaround -- at the time of writing, determinate-nixd
```

Comment:
nit: does it make sense to move these closer to where you actually define/set the duration? (for both of these pattern)